benchmarks/devel/devel.cpp

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_DetailsUtils.hpp> // accumulate
+#include <ArborX_Version.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <chrono>
+
+#include <benchmark/benchmark.h>
+
+template <typename ExecutionSpace>
+auto setup(ExecutionSpace const &exec_space, int n)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+  Kokkos::View<int *, MemorySpace> view("Benchmark::view", n);
+  return view;
+}
+
+template <typename ExecutionSpace, typename View>
+void kernel(ExecutionSpace const &exec_space, View view)
+{
+  ArborX::accumulate(exec_space, view, 0);
+}
+
+template <typename ExecutionSpace>
+void BM_benchmark(benchmark::State &state, int n)
+{
+  ExecutionSpace exec_space;
+
+  auto view = setup(exec_space, n);
+
+  for (auto _ : state)
+  {
+    exec_space.fence();
+    auto const start = std::chrono::high_resolution_clock::now();
+
+    kernel(exec_space, view);
+
+    exec_space.fence();
+    auto const end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed_seconds = end - start;
+    state.SetIterationTime(elapsed_seconds.count());
+  }
+  state.counters["rate"] =
+      benchmark::Counter(n, benchmark::Counter::kIsIterationInvariantRate);
+}
+
+int main(int argc, char *argv[])
+{
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  std::cout << "ArborX version    : " << ArborX::version() << std::endl;
+  std::cout << "ArborX hash       : " << ArborX::gitCommitHash() << std::endl;
+  std::cout << "Kokkos version    : " << KokkosExt::version() << std::endl;
+
```

Comment:
So we don't have to relink every time.  You can get the kokkos version by passing `--kokkos-print-configuration` or setting `KOKKOS_PRINT_CONFIGURATION=1`.  I don't think this belong to a minimal benchmark in the first place.
```suggestion
```

Reply:
I very much disagree. I would never want to run a benchmark without having the version details. It is being very careful with benchmarking, and being able to match some of the logs back without relying on memory to include the options. 

I understand the point about relinking. However, I don't see that as a big problem.